addon/merge/merge.js

```diff
@@ -5,15 +5,18 @@
 
 (function(mod) {
   if (typeof exports == "object" && typeof module == "object") // CommonJS
-    mod(require("../../lib/codemirror"), require("diff_match_patch"));
+    mod(require("../../lib/codemirror"), require("diff-match-patch"));
   else if (typeof define == "function" && define.amd) // AMD
     define(["../../lib/codemirror", "diff_match_patch"], mod);
   else // Plain browser env
-    mod(CodeMirror);
-})(function(CodeMirror) {
+    mod(CodeMirror, diff_match_patch);
+})(function(CodeMirror, diff_match_patch) {
   "use strict";
   var Pos = CodeMirror.Pos;
   var svgNS = "http://www.w3.org/2000/svg";
+  var DIFF_INSERT = diff_match_patch.DIFF_INSERT || DIFF_INSERT;
+  var DIFF_DELETE = diff_match_patch.DIFF_DELETE || DIFF_DELETE;
+  var DIFF_EQUAL = diff_match_patch.DIFF_EQUAL || DIFF_EQUAL;
```

Comment:
ugh yes. thanks.

Also noticed that these enum values are simple ints, where `DIFF_EQUAL` is
0 in the original implementation, so the `||` shortcircuiting will fail in
the last case, where `diff_match_patch.DIFF_EQUAL` is 0 and `DIFF_EQUAL` is
undefined.

On Wed, Oct 14, 2015 at 6:30 AM, Forbes Lindesay <notifications@github.com>
wrote:

> In addon/merge/merge.js
> <https://github.com/codemirror/CodeMirror/pull/3593#discussion_r41992385>:
>
> >    "use strict";
> >    var Pos = CodeMirror.Pos;
> >    var svgNS = "http://www.w3.org/2000/svg";
> > +  var DIFF_INSERT = diff_match_patch.DIFF_INSERT || DIFF_INSERT;
> > +  var DIFF_DELETE = diff_match_patch.DIFF_DELETE || DIFF_DELETE;
> > +  var DIFF_EQUAL = diff_match_patch.DIFF_EQUAL || DIFF_EQUAL;
>
> The || DIFF_EQUAL here won't work because you just declared DIFF_EQUAL so
> it will always be undefined, You either need to make your variable have a
> different name, or get a reference to the global object first and then do
> something like || window.DIFF_EQUAL.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codemirror/CodeMirror/pull/3593/files#r41992385>.
>